pkg/validation/options.go

```diff
@@ -411,3 +417,71 @@ func parseURL(toParse string, urltype string, msgs []string) (*url.URL, []string
 	}
 	return parsed, msgs
 }
+
+// CallbackErrorRedirect - Allow automated redirection to occur within the AuthCallback when an error occurs
+type CallbackErrorRedirect struct {
+	ErrorString string
+	Pattern     string
+	RedirectURL string
+}
+
+func loadCERsExtractFromRegexp(pat *regexp.Regexp, match []string, pcer *CallbackErrorRedirect, perr *error) {
+	for i, name := range pat.SubexpNames() {
+		if i == 0 || name == "" {
+			continue
+		}
+
+		switch name {
+		case "error_string":
+			if pcer.ErrorString == "" {
+				pcer.ErrorString = match[i]
+			} else if *perr == nil {
+				*perr = errors.New("callback error redirect: error_string was specified more than once")
+			}
+		case "error_description":
+			if pcer.Pattern == "" {
+				pcer.Pattern = match[i]
+			} else if *perr == nil {
+				*perr = errors.New("callback error redirect: error_description was specified more than once")
+			}
+		case "redirect":
+			if pcer.RedirectURL == "" {
+				pcer.RedirectURL = match[i]
+			} else if *perr == nil {
+				*perr = errors.New("callback error redirect: redirect was specified more than once")
+			}
+		}
+	}
+}
+
+//LoadCERs - Convert the string versions of callback-error-redirect into structures
+func LoadCERs(callbackErrorRedirects []string) ([]CallbackErrorRedirect, error) {
+
+	//error_string=foo,error_description=bar,redirect=baz
+
+	pat := regexp.MustCompile(`^\s*((error_string\s*=\s*(?P<error_string>[^,$]*)|error_description\s*=\s*(?P<error_description>[^,$]*)|redirect\s*=\s*(?P<redirect>[^,$]*)),?)+\s*`)
```

Comment:
Does this allow the three options to be reordered at all? Is it preferable to allow them to be reordered?

Reply:
Short answer - they can go in any order, and I think that it's better that way.

Reply:
SGTM